eng/pipelines/templates/jobs/archetype-sdk-client.yml

```diff
@@ -195,7 +195,8 @@ jobs:
 
   - script: |
       make az_core_cov_xml
-      make az_iot_cov_xml
+      make az_iot_hub_cov_xml
+      make az_iot_pnp_cov_xml
```

Comment:
How is this xml name generated? Is it based on some convention tied to the CMakeLists.txt?
https://github.com/Azure/azure-sdk-for-c/blob/06fa163726dafe26838fc0fecb755976637ba6c0/sdk/iot/pnp/CMakeLists.txt#L6

Do we take the target name and add `_cov_xml` to it? Maybe we should add a comment there?

Reply:
yes, but we have a little script in cmake-modules that makes the job:
```cmake
# codeCoverage
if(DEFINED ENV{AZ_SDK_CODE_COV} AND CMAKE_C_COMPILER_ID MATCHES "GNU")
    if(CMAKE_BUILD_TYPE STREQUAL "Debug")
        APPEND_COVERAGE_COMPILER_FLAGS()
        
        # Basic coverage using lcov (gcc integrated)
        setup_target_for_coverage_lcov(NAME ${TARGET_NAME}_cov
                                       EXECUTABLE ${TARGET_NAME}_test
                                       EXCLUDE ${COV_EXCLUDE})
        
        # HTML and XML - Coverage using gcovr (Needs to be installed into system)
        setup_target_for_coverage_gcovr_html(NAME ${TARGET_NAME}_cov_html EXECUTABLE ${TARGET_NAME}_test)
        setup_target_for_coverage_gcovr_xml(NAME ${TARGET_NAME}_cov_xml EXECUTABLE ${TARGET_NAME}_test)
    endif() 
endif()
```

projects just need to include this